vm/src/stdlib/bisect.rs

```diff
@@ -0,0 +1,182 @@
+pub(crate) use _bisect::make_module;
+
+#[pymodule]
+mod _bisect {
+    use std::convert::TryFrom;
+
+    use crate::function::OptionalArg;
+    use crate::slots::PyComparisonOp::Lt;
+    use crate::vm::VirtualMachine;
+    use crate::{ItemProtocol, PyObjectRef, PyResult};
+    use num_traits::ToPrimitive;
+
+    #[derive(FromArgs)]
+    struct BisectArgs {
+        #[pyarg(any, name = "a")]
+        a: PyObjectRef,
+        #[pyarg(any, name = "x")]
+        x: PyObjectRef,
+        #[pyarg(any, optional, name = "lo")]
+        lo: OptionalArg<PyObjectRef>,
+        #[pyarg(any, optional, name = "hi")]
+        hi: OptionalArg<PyObjectRef>,
+    }
+
+    // Handles objects that implement __index__ and makes sure index fits in needed isize.
+    #[inline]
+    fn handle_default(
+        arg: OptionalArg<PyObjectRef>,
+        default: Option<isize>,
+        vm: &VirtualMachine,
+    ) -> PyResult<Option<isize>> {
+        Ok(match arg {
+            OptionalArg::Present(v) => {
+                Some(vm.to_index(&v)?.as_bigint().to_isize().ok_or_else(|| {
+                    vm.new_index_error("cannot fit 'int' into an index-sized integer".to_owned())
+                })?)
+            }
+            OptionalArg::Missing => default,
+        })
+    }
+
+    // Handles defaults for lo, hi.
+    //
+    //  - lo must be >= 0 with a default of 0.
+    //  - hi, while it could be negative, defaults to `0` and the same effect is achieved
+    //    (while loop isn't entered); this way we keep it a usize (and, if my understanding
+    //    is correct, issue 13496 is handled). Its default value is set to the length of the
+    //    input sequence.
+    #[inline]
+    fn as_usize(
+        lo: OptionalArg<PyObjectRef>,
+        hi: OptionalArg<PyObjectRef>,
+        seq_len: usize,
+        vm: &VirtualMachine,
+    ) -> PyResult<(usize, usize)> {
+        // We only deal with positives for lo, try_from can't fail.
+        // Default is always a Some so we can safely unwrap.
+        let lo = handle_default(lo, Some(0), vm)?
+            .map(|value| {
+                if value < 0 {
+                    return Err(vm.new_value_error("lo must be non-negative".to_owned()));
+                }
+                Ok(usize::try_from(value).unwrap())
```

Comment:
Instead of unwrapping the result here, you could do `usize::try_from(value).map_err(|_| vm.new_error...)` - `usize::try_from(isize)` only fails when the isize is negative.